cpp/src/arrow/engine/substrait/util.h

```diff
@@ -39,6 +39,9 @@ ARROW_ENGINE_EXPORT Result<std::shared_ptr<RecordBatchReader>> ExecuteSerialized
 ARROW_ENGINE_EXPORT Result<std::shared_ptr<Buffer>> SerializeJsonPlan(
     const std::string& substrait_json);
 
+ARROW_ENGINE_EXPORT Result<std::vector<compute::Declaration>> DeserializePlans(
+    const Buffer& buf, const ExtensionIdRegistry* registry);
```

Comment:
There are already functions named `DeserializePlans` in `serde.h`. Isn't it a bit confusing to have another one similarly named here?

Also, can you add a docstring?

Reply:
The main purpose of `util.h` is to expose functions to PyArrow. Nevertheless, I'll go over these function to see what can be simplified,